src/Runner.Listener/Runner.cs

```diff
@@ -494,8 +494,28 @@ private async Task<int> RunAsync(RunnerSettings settings, bool runOnce = false)
                                         }
                                     }
 #endif
-                                    var selfUpdater = HostContext.GetService<ISelfUpdater>();
-                                    selfUpdateTask = selfUpdater.SelfUpdate(runnerUpdateMessage, jobDispatcher, false, HostContext.RunnerShutdownToken);
+                                    if (brokerRunnerUpdateMessage != null)
+                                    {
+                                        var selfUpdater = new SelfUpdaterV2();
+                                        selfUpdater.Initialize(HostContext);
+                                        var refreshMessage = new AgentRefreshMessage(brokerRunnerUpdateMessage.RunnerId, brokerRunnerUpdateMessage.TargetVersion, TimeSpan.FromSeconds(brokerRunnerUpdateMessage.TimeoutInSeconds))
```

Comment:
Why do we treat most of the RefreshMessage's values as readonly, but not package metadata?

Reply:
updated